napari/components/viewer_model.py

```diff
@@ -1053,19 +1059,6 @@ def _open_or_raise_error(
                     plugin,
                     paths,
                 ) from e
-
-        # preferred plugin doesn't exist
-        elif plugin:
-            raise MissingAssociatedReaderError(
-                trans._(
-                    "Can't find {plugin} plugin associated with {extension} files.",
```

Comment:
if this specifically arises in the case where the plugin had been stored in the preferences, then I think the error message could make it clearer that this is a preferences/environment conflict?  am I understanding this correctly?

Also, maybe I missed it, but is there anywhere that we're either removing the association from preferences, or instructing a user on what they might do to fix the error?  or will it just pop up repeatedly?

Reply:
Good point @tlambert03 thanks for flagging. We don't edit the user's preferences - I've updated the wording of the warning to let users know how they can remove the preference. However since this might just be a case of "oh shoot, I'm in the wrong environment" I don't think we should be editing the preference for them.

Note that in the GUI if this warning were thrown and multiple plugins were *still* available, they would get both the warning and the dialog to select a new plugin.

Does this seem reasonable? I know you're worried about throwing more errors. I am too, even though I do actually think it's the correct path forward. This one is a warning 😛 , but still - let me know if you've still got concerns.